src/main/java/com/amazon/opendistroforelasticsearch/security/securityconf/ConfigModelV6.java

```diff
@@ -1100,47 +1100,52 @@ public TenantHolder(SecurityDynamicConfiguration<RoleV6> roles) {
     private class RoleMappingHolder {
 
         private ListMultimap<String, String> users;
-        private ListMultimap<Set<String>, String> abars;
+        private ListMultimap<Collection<WildcardMatcher>, String> abars;
```

Comment:
nit: not sure if `Collection` is the best option here. If it is always `List`, should we use `List` instead?

Reply:
Not for this PR, but using `Collection`/`List`/`Set` of `WildcardMatcher` as key does not sound right to me. IMO, this requires more analysis and likely a follow up PR.

Reply:
Created a follow-up issue https://github.com/opendistro-for-elasticsearch/security/issues/906